cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -163,7 +163,12 @@ func (scaleSet *ScaleSet) Autoprovisioned() bool {
 func (scaleSet *ScaleSet) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
 	template, err := scaleSet.getVMSSFromCache()
 	if err != nil {
-		return nil, err
+		klog.Errorf("failed to get information for VMSS: %s", scaleSet.Name)
+		// Note: We don't return an error here and instead accept defaults.
+		// Every invocation of GetOptions() returns an error if this condition is met:
+		// `if err != nil && err != cloudprovider.ErrNotImplemented`
+		// The error return value is intended to only capture unimplemented.
+		return nil, nil
```

Comment:
What do you think about returning `cloudprovider.ErrNotImplemented` instead?
It is true that we don't implement this, and the handling will be the same as just returning `nil, nil`. Just with clearer visual.

Reply:
Hm... I don't think that's exactly accurate. `ErrNotImplemented` signifies that we don't implement GetOptions() at all which isn't the case. We just were unable to get options, because we were unable to get the scaleset. 

The function comment here explains what nil will do:
>  Returning a nil will result in using default options.